### ISSUE TITLE
Multiple Builder pattern support

### ISSUE THREAD

**Issue by fborriello** [HotelsDotCom/bull#144]
**Description**

As per today, the `BeanTransformer` is able to transform Java Bean using the following builder pattern:

```
public class ItemType {
    private final Class<?> objectClass;
    private final Class<?> genericClass;

    ItemType(final Class<?> objectClass, final Class<?> genericClass) {
        this.objectClass = objectClass;
        this.genericClass = genericClass;
    }

    public static ItemTypeBuilder builder() {
        return new ItemType.ItemTypeBuilder();
    }

    // getter methods

    public static class ItemTypeBuilder {
        private Class<?> objectClass;
        private Class<?> genericClass;

        ItemTypeBuilder() {
        }

        public ItemTypeBuilder objectClass(final Class<?> objectClass) {
            this.objectClass = objectClass;
            return this;
        }

        public ItemTypeBuilder genericClass(final Class<?> genericClass) {
            this.genericClass = genericClass;
            return this;
        }

        public com.hotels.transformer.model.ItemType build() {
            return new ItemType(this.objectClass, this.genericClass);
        }
    }
}
```

The purpose is to support also the following pattern too:

```
public class ItemType {
    private final Class<?> objectClass;
    private final Class<?> genericClass;

    ItemType(final ItemTypeBuilder builder) {
        this.objectClass = builder.objectClass;
        this.genericClass = builder.genericClass;
    }

    public static ItemTypeBuilder builder() {
        return new ItemType.ItemTypeBuilder();
    }

    // getter methods

    public static class ItemTypeBuilder {
        private Class<?> objectClass;
        private Class<?> genericClass;

        ItemTypeBuilder() {
        }

        public ItemTypeBuilder objectClass(final Class<?> objectClass) {
            this.objectClass = objectClass;
            return this;
        }

        public ItemTypeBuilder genericClass(final Class<?> genericClass) {
            this.genericClass = genericClass;
            return this;
        }

        public com.hotels.transformer.model.ItemType build() {
            return new ItemType(this);
        }
    }
}
```

**Acceptance Criteria**

* **The solution must not degrade the performances**
* The `BeanTransformer` automatically recognized the Builder pattern and use it properly

**Comment by fborriello** [comment]
Release candidate available for testing: [1.7.0-RC](https://github.com/HotelsDotCom/bull/releases/tag/1.7.0-RC)

**Comment by fborriello** [comment]
## How it works?

Java Bean with the Standard Builder pattern are transformed natively, but to enable it for Java Beans using a **custom Builder pattern**, It's needed to enable the feature as following:

```
ToBean toBean = new BeanTransformer()
                         .setCustomBuilderTransformationEnabled(true)
                         .transform(sourceObject, ToBean.class);
```

**Comment by fborriello** [comment]
Feature available from version: `1.7.0.0` (for `jdk 11`) and `1.7.0-jdk8` (for `jdk 8`)
